Makefile

```diff
@@ -1,33 +1,53 @@
+SWAGGER_CODEGEN_JAR := http://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.1.6/swagger-codegen-cli-2.1.6.jar
+STUB_SERVER_FOLDER := edx-api-stub-server
+
 help:
-	@echo '                                                                                     		'
-	@echo 'Makefile for api-manager                                                                 	'
-	@echo '                                                                                     		'
-	@echo 'Usage:                                                                               		'
-	@echo '    make requirements                 install requirements for local development     		'
-	@echo '    make clean                        delete generated byte code                 		'
-	@echo '    make quality                      run PEP8 and Pylint                            		'
-	@echo '    make validate                     Run Python unit tests and linting 	                	'
-	@echo '                                                                                     		'
+	@echo '                                                                                     '
+	@echo 'Makefile for api-manager                                                             '
+	@echo '                                                                                     '
+	@echo 'Usage:                                                                               '
+	@echo '    make clean                Delete generated python byte code and testing remnants '
+	@echo '    make requirements         Install requirements for local development             '
+	@echo '    make quality              Run PEP8 and Pylint                                    '
+	@echo '    make build                Flatten the swagger docs and build a stub server       '
+	@echo '    make test                 Run all tests that do not require a running server     '
+	@echo '    make run                  Run a stub server that serves the swagger docs         '
+	@echo '                                                                                     '
+
+clean:
+	@find . -name '*.pyc .cache __pycache__' -delete
 
 requirements:
-	pip install -qr requirements/test.txt --exists-action w
+	@pip install -qr requirements/test.txt --exists-action w
 
-clean:
-	find . -name '*.pyc' -delete
+quality:
+	@pep8 --config=.pep8 scripts/aws
+	@pylint --rcfile=.pylintrc scripts/aws
 
-test_scripts:
-	cd scripts/aws && python -m pytest
+clean.stub:
+	@rm -rf $(STUB_SERVER_FOLDER)
 
-validate_python:
-	make clean
-	make quality
-	make test_scripts
+# Download the swagger codegen jar if you don't already have it.
+# TODO: verify via checksum that the file is valid.
```

Comment:
@efagin Still thinking through how best to do the checksum that will work on either OSX or ubuntu. Open to suggestions!